readthedocs/rtd_tests/tests/test_config_integration.py

```diff
@@ -199,23 +211,582 @@ def test_conda(self, load_config):
+        self.assertEqual(config.python.requirements, '__init__.py')
+
+
+@pytest.mark.skip
+@pytest.mark.django_db
+@mock.patch('readthedocs.projects.models.Project.checkout_path')
+class TestLoadConfigV2(object):
+
+    @pytest.fixture(autouse=True)
+    def create_project(self):
+        self.project = get(
+            Project,
+            main_language_project=None,
+            install_project=False,
+            container_image=None,
+        )
+        self.version = get(Version, project=self.project)
+        # TODO: Remove later
+        get(
+            Feature,
+            projects=[self.project],
+            feature_id=Feature.ALLOW_V2_CONFIG_FILE,
+        )
+
+    def create_config_file(self, tmpdir, config):
+        base_path = apply_fs(tmpdir, {
+            'readthedocs.yml': '',
+        })
+        config.setdefault('version', 2)
+        config_file = path.join(str(base_path), 'readthedocs.yml')
+        yaml.safe_dump(config, open(config_file, 'w'))
+        return base_path
+
+    def get_update_docs_task(self):
+        build_env = LocalBuildEnvironment(
+            self.project, self.version, record=False
+        )
+
+        update_docs = tasks.UpdateDocsTaskStep(
+            build_env=build_env,
+            config=load_yaml_config(self.version),
+            project=self.project,
+            version=self.version,
+        )
+        return update_docs
+
+    def test_using_v2(self, checkout_path, tmpdir):
+        checkout_path.return_value = str(tmpdir)
+        self.create_config_file(tmpdir, {})
+        update_docs = self.get_update_docs_task()
+        assert update_docs.config.version == '2'
+
+    def test_report_using_invalid_version(self, checkout_path, tmpdir):
+        checkout_path.return_value = str(tmpdir)
+        self.create_config_file(tmpdir, {'version': 12})
+        with pytest.raises(InvalidConfig) as exinfo:
+            self.get_update_docs_task()
+        assert exinfo.value.key == 'version'
+
+    @pytest.mark.parametrize('config', [{}, {'formats': []}])
+    @patch('readthedocs.projects.models.Project.repo_nonblockinglock', new=MagicMock())
+    @patch('readthedocs.doc_builder.backends.sphinx.SearchBuilder.build')
+    @patch('readthedocs.doc_builder.backends.sphinx.HtmlBuilder.build')
+    @patch('readthedocs.doc_builder.backends.sphinx.HtmlBuilder.append_conf')
+    def test_build_formats_default_empty(
+            self, append_conf, html_build, search_build,
+            checkout_path, config, tmpdir):
+        """
+        The default value for formats is [], which means no extra
+        formats are build.
+        """
+        checkout_path.return_value = str(tmpdir)
+        self.create_config_file(tmpdir, config)
+
+        update_docs = self.get_update_docs_task()
+        outcomes = update_docs.build_docs()
+
+        # No extra formats were triggered
+        assert outcomes['html']
+        assert outcomes['search']
+        assert not outcomes['localmedia']
```

Comment:
`htmlzip` is missing and it's a valid option.

Reply:
`htmlzip` is `localmedia` https://github.com/stsewd/readthedocs.org/blob/build-config-v2-implementation/readthedocs/projects/tasks.py#L744-L752

Reply:
This was the feature that we want to implement a time ago but it was already there :laughing: